Source/Microsoft.Teams.Apps.FAQPlusPlus.Common/Providers/QnaServiceProvider.cs

```diff
@@ -183,19 +183,29 @@ public async Task DeleteQnaAsync(int questionId)
         /// </summary>
         /// <param name="question">Question text.</param>
         /// <param name="isTestKnowledgeBase">Prod or test.</param>
-        /// <returns>QnaSearchResult result as response.</returns>
-        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase)
+        /// <param name="payload">The response card payload which contains the previous question when querying follow up prompts.</param>
+        /// <returns>QnaSearchResultList result as response.</returns>
+        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase, ResponseCardPayload payload = null)
         {
             var knowledgeBaseId = await this.configurationProvider.GetSavedEntityDetailAsync(ConfigurationEntityTypes.KnowledgeBaseId).ConfigureAwait(false);
 
-            QnASearchResultList qnaSearchResult = await this.qnaMakerRuntimeClient.Runtime.GenerateAnswerAsync(knowledgeBaseId, new QueryDTO()
+            QueryDTO queryDTO = new QueryDTO()
             {
                 IsTest = isTestKnowledgeBase,
                 Question = question?.Trim(),
                 ScoreThreshold = Convert.ToDouble(this.options.ScoreThreshold),
-            }).ConfigureAwait(false);
+            };
+
+            if (payload != null && payload.IsPrompt)
+            {
+                queryDTO.Context = new QueryDTOContext
+                {
+                    PreviousQnaId = payload.PreviousQuestions.FirstOrDefault().Id.ToString(),
+                    PreviousUserQuery = payload.PreviousQuestions.FirstOrDefault().Questions.FirstOrDefault(),
```

Comment:
This seems to be setting it to the Question associated with the previous answer, which is not quite the same as the previous user query, right?
E.g., if my question is "sign in", and that matched "How do I log in to my account?", does qnamaker expecct PreviousUserQuery to be "sign in", or "How do I log in to my account?" (Or does it not matter?)

Reply:
The value of previous user query will be "How do I log in to my account?".

Reply:
Right, which one does QnA Maker expect-- the actual query typed by the user, or the matched question (even if that was not what was entered). Can we clarify please? Thanks!

Reply:
QnA Maker expects the matched question.
E.g., if user question is "sign in" and the matched question returned is "How do I sign in to my account?", then the qnamaker expects PreviousUserQuery to be the matched question "How do I sign in to my account?".